processor/transformprocessor/README.md

```diff
@@ -33,6 +33,10 @@ Supported functions:
 e.g., `set(attributes["http.path"], "/foo")`, `set(name, attributes["http.route"])`, `set(trace_state["svc"], "example")`, `set(attributes["source"], trace_state["source"])`. If `value` resolves to `nil`, e.g.
 it references an unset map value, there will be no action.
 
+- `delete_key(target, key)` - `target` is a path expression to a map type field. `key` is a string that is a key in the map.  The key will be deleted from the map.  e.g., `delete_key(attributes, "http.request.header.authorization")`
+
+- `delete_matching_keys(target, pattern)` - `target` is a path expression to a map type field. `pattern` is a regex string.  All keys that match the pattern will be deleted from the map.  e.g., `delete_matching_keys(attributes, ".*\.header\.authorization")`
```

Comment:
Should this be `delete_all_matches` to be consistent with `replace_all_matches`?

Reply:
I thought about that but I didn't want anyone to think it deleted the matching sections in the map's value vs replacing it like `replace_all_matches`.  I wanted it to be clear that its the whole entry being deleted, so I kept it closer to `delete_key`